session/sessionstate_test.go

```diff
@@ -403,3 +404,42 @@ func TestStateTemplateArtifactMap(t *testing.T) {
 	})
 
 }
+
+func TestPendingWaiter(t *testing.T) {
+	counters := &statistics.ExecutionCounters{}
+	state := New(context.Background(), "", 120, nil, 1, 1, "", false, counters)
```

Comment:
Did this simple test break the old one? If so, good! If not it would be a good idea to create such a test.